text_generation/causal_lm/cpp/greedy_causal_lm.cpp

```diff
@@ -1,129 +1,27 @@
-    // Input values are persistent between inference calls.
-    // That allows to set values, which aren't going to change, only once
-    lm.get_tensor("beam_idx").set_shape({BATCH_SIZE});
-    lm.get_tensor("beam_idx").data<int32_t>()[0] = 0;
-    lm.infer();
-    size_t vocab_size = lm.get_tensor("logits").get_shape().back();
-    float* logits = lm.get_tensor("logits").data<float>() + (seq_len - 1) * vocab_size;
-    int64_t out_token = std::max_element(logits, logits + vocab_size) - logits;
-
-    lm.get_tensor("input_ids").set_shape({BATCH_SIZE, 1});
-    position_ids.set_shape({BATCH_SIZE, 1});
-    TextStreamer text_streamer{std::move(detokenizer)};
-
-    // Get the runtime info from the tokenizer model that we read earlier
-    auto rt_info = tokenizer_model->get_rt_info(); //Get the runtime info for the model
-    int64_t SPECIAL_EOS_TOKEN;
+    if (3 > argc || argc > 4)
+        throw std::runtime_error(std::string{"Usage: "} + argv[0] + " <MODEL_DIR> \"<PROMPT>\" <DEVICE>");
 
-    if (rt_info.count("eos_token_id") > 0) { //check if the runtime information has a valid EOS token ID
-        SPECIAL_EOS_TOKEN = rt_info["eos_token_id"].as<int64_t>();
-    } else {
-        throw std::runtime_error("EOS token ID not found in model's runtime information.");
-    }
-
-    int max_sequence_length = 100;
-    while (out_token != SPECIAL_EOS_TOKEN && seq_len < max_sequence_length) {
-        ++seq_len;
-        lm.get_tensor("input_ids").data<int64_t>()[0] = out_token;
-        lm.get_tensor("attention_mask").set_shape({BATCH_SIZE, seq_len});
-        std::fill_n(lm.get_tensor("attention_mask").data<int64_t>(), seq_len, 1);
-        position_ids.data<int64_t>()[0] = int64_t(seq_len - 1);
-        lm.start_async();
-        text_streamer.put(out_token);
-        lm.wait();
-        logits = lm.get_tensor("logits").data<float>();
-        out_token = std::max_element(logits, logits + vocab_size) - logits;
-    }
-    text_streamer.end();
-    // Model is stateful which means that context (kv-cache) which belongs to a particular
-    // text sequence is accumulated inside the model during the generation loop above.
-    // This context should be reset before processing the next text sequence.
-    // While it is not required to reset context in this sample as only one sequence is processed,
-    // it is called for education purposes:
-    lm.reset_state();
+    std::string model_path = argv[1];
+    std::string prompt = argv[2];
+    
+    // GPU can be used as well
+    std::string device = "CPU";  
+    if (argc > 3) device = argv[3];
+
+    ov::LLMPipeline pipe(model_path, device);
+    ov::GenerationConfig config = pipe.get_generation_config();
+    config.max_new_tokens = 100;
+    config.do_sample = false;
+    auto streamer = [](std::string subword){std::cout << subword << std::flush;};
+    
+    // since streamer is set results will be printed each time a new token is generated
+    pipe.generate(prompt, config, streamer);
```

Comment:
@Wovchena could you please also help later to add Python sample similar to C++ ones?